editor/plugins/animation_player_editor_plugin.cpp

```diff
@@ -1091,9 +1091,30 @@ void AnimationPlayerEditor::_animation_duplicate() {
 		return;
 	}
 
-	String new_name = current;
-	while (player->has_animation(new_name)) {
-		new_name = new_name + " (copy)";
+	int index_digits = 0;
+	for (int i = current.length() - 1; i >= 0; i--) {
+		if (is_digit(current[i])) {
+			index_digits++;
+		} else {
+			break;
+		}
+	}
+	int64_t count = 2;
+	if (index_digits) {
+		count = current.right(index_digits).to_int();
+	}
+	String new_name = (index_digits) ? current.left(-index_digits) : current + "_";
```

Comment:
If there is a `new_animation` and `new_animation2`, they will be duplicated under the name new_animation3, which violates the godot style. So what happens if you then create `new_animation` again?

It will create `new_animation2`, then there are `new_animation2` and `new_animation_2`, which will be confusing.

So in my opinion, it absolutely should be parsed indexed by underscore and renamed as `new_animation2_2`.

Reply:
For a new_animation2 to exist in this workflow, it means the user duplicated a new_animation, then erased the _ between new_animation and 2.

Reply:
This is to prevent inconsistencies in the core code.